utils/s2n_init.c

```diff
@@ -89,9 +89,10 @@ static bool s2n_cleanup_atexit_impl(void)
     /* the configs need to be wiped before resetting the memory callbacks */
     s2n_wipe_static_configs();
 
-    return s2n_result_is_ok(s2n_libcrypto_cleanup()) &&
+    return s2n_result_is_ok(s2n_cipher_suites_cleanup()) &&
         s2n_result_is_ok(s2n_rand_cleanup_thread()) &&
         s2n_result_is_ok(s2n_rand_cleanup()) &&
+        s2n_result_is_ok(s2n_libcrypto_cleanup()) &&
```

Comment:
why move `s2n_libcrypto_cleanup` below rand_cleanup?

Since we call init in the follow order we want to keep cleanup in the reverse order:  

```
    POSIX_GUARD_RESULT(s2n_rand_init());
    POSIX_GUARD(s2n_cipher_suites_init());
```

Reply:
Please check s2n_init again, you will find:

1. s2n_libcrypto_init is called before s2n_rand_init, therefore in the cleanup routine it should be called after s2n_rand_cleanup - previously it was being called before, which would be out of order, so I corrected that.  That is the change shown above.

2. s2n_cipher_suites_init is called after s2n_rand_init, therefore in the cleanup routine it should be called before s2n_rand_cleanup, which is what I did (it is first).